presto-hive-common/src/main/java/com/facebook/presto/hive/CacheQuota.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import io.airlift.units.DataSize;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.hash.Hashing.md5;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+
+public class CacheQuota
+{
+    public static final CacheQuota NO_CACHE_CONSTRAINTS = new CacheQuota("NO_IDENTITY", Optional.empty());
+
+    private final long identifier;
+    private final Optional<DataSize> quota;
+
+    public CacheQuota(String identity, Optional<DataSize> quota)
+    {
+        this.identifier = md5().hashString(identity, UTF_8).asLong();
```

Comment:
I thought we gonna move this to `getIdentifier()` method?

Reply:
> I thought we gonna move this to `getIdentifier()` method?

I have changed the design, so now this happens on workers now